src/lazyloading.js

```diff
@@ -1,53 +1,136 @@
 /* eslint-disable global-require */
 /* eslint-disable no-param-reassign */
-export default {
-    inserted: (el) => {
-        const { style } = el.dataset;
-        const img = el.dataset.image;
-        const className = el.getAttribute('class');
-        const src = el.getAttribute('src');
-        el.setAttribute('class', 's-image');
-        el.setAttribute('style', !style ? 'height: 170px;' : style);
-        el.setAttribute('src', !img ? require('./assets/img/img_skeleton.png') : img);
 
-        const loadImage = () => {
-            el.src = src;
-        };
+/**
+ * Data image default
+ * {Object}
+ * */
+const imageDefault = {
+    src: require('./assets/img/img_skeleton.png'),
+    style: 'height: 170px',
+    class: 's-image',
+};
+/**
+ * Observer options default
+ * {Object}
+ * */
+const observerDefault = {
+    root: null,
+    threshold: 1,
+};
 
-        const callback = (entries, observer) => {
-            entries.forEach((entry) => {
-                if (entry.isIntersecting) {
-                    loadImage();
-                    observer.unobserve(el);
-                }
-            });
-        };
+export default {
+  name: 'CoreLazyLoading',
+  props: {
+    placeholderImage: {
+      type: Object,
+      default: () => {
+        return imageDefault
+      },
+    },
+
+    options: {
+      type: Object,
+      default: () => {
+        return observerDefault
+      },
+    },
+  },
+  data() {
+    return {
+      observer: {},
```

Comment:
Where do you use it?

Reply:
![image](https://user-images.githubusercontent.com/59548686/104023515-9ab65a00-51e3-11eb-9742-261c295c2dac.png)
![image](https://user-images.githubusercontent.com/59548686/104023533-a013a480-51e3-11eb-9e9c-d17daaaca872.png)

Reply:
But you are not calling it as `this.observer`. Look at your code, each time you are initializing a new variable with this name and you haven't used this parameter from your data object.